homeassistant/components/here_weather/__init__.py

```diff
@@ -0,0 +1,143 @@
+"""The here_weather component."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+import herepy
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_SCAN_INTERVAL,
+    CONF_UNIT_SYSTEM_METRIC,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_MODES, DEFAULT_SCAN_INTERVAL, DOMAIN, HERE_API_KEYS
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor", "weather"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up here_weather from a config entry."""
+    here_weather_data_dict = {}
+    for mode in CONF_MODES:
+        here_weather_data = HEREWeatherData(hass, entry, mode)
+        await here_weather_data.async_setup()
+        here_weather_data_dict[mode] = here_weather_data
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = here_weather_data_dict
+
+    known_api_keys = hass.data.setdefault(HERE_API_KEYS, [])
```

Comment:
No need to store the known keys in hass.data. We can check the keys of the existing config entries.

Reply:
Done in https://github.com/home-assistant/core/pull/28910/commits/34104ed8240aef0e98e90517905aadf757958d97